coderd/database/queries/templates.sql

```diff
@@ -105,3 +105,56 @@ WHERE
 	id = $1
 RETURNING
 	*;
+
+-- name: GetTemplatesAverageBuildTime :many
+WITH query_with_all_job_count AS (SELECT
```

Comment:
I don't understand why we need to calculate a moving average here. Can we take the last 10 builds and average the times? I'm trying to see if there's a simpler query to what we want to achieve.

Reply:
@ammario, how else would you keep 10 rows per group?

Reply:
I believe since we just need the average of the last 10 builds, we can combine a `GROUP BY template_id` with an `AVERAGE(exec_time_sec)` to compute the average. We will still need all the joins, though. We can merge the PR without this though, it's not a severe enough matter.

Reply:
@ammario, how would you pick up the last N builds with a group by without any sort of ranking?

Reply:
You would do the limit + order by date DESC in a subquery, then aggregate in the higher level query. In the higher level query all the rows apply, so no analytic/window function should be necessary, just pure aggregation.

Reply:
You can't apply LIMIT per group, only a global one. Here we want N builds per template.

Reply:
We discussed over Slack and I was mistaken. The query looks good.